lib/matplotlib/font_manager.py

```diff
@@ -1332,22 +1332,23 @@ def get_font(filename, hinting_factor=None):
                      _kerning_factor=rcParams['text.kerning_factor'])
 
 
-def _rebuild():
-    global fontManager
-    _log.info("Generating new fontManager, this may take some time...")
-    fontManager = FontManager()
-    json_dump(fontManager, _fmcache)
-
-
-try:
-    fontManager = json_load(_fmcache)
-except Exception:
-    _rebuild()
-else:
-    if getattr(fontManager, '_version', object()) != FontManager.__version__:
-        _rebuild()
-    else:
-        _log.debug("Using fontManager instance from %s", _fmcache)
+def _load_fontmanager(*, try_read_cache=True):
+    if try_read_cache:
+        fm_path = Path(
+            mpl.get_cachedir(), f"fontlist-v{FontManager.__version__}.json")
+        try:
+            fm = json_load(fm_path)
+        except Exception as exc:
+            pass
+        else:
+            if getattr(fm, "_version", object()) == FontManager.__version__:
+                _log.debug("Using fontManager instance from %s", fm_path)
+                return fm
+    fm = FontManager()
+    json_dump(fm, fm_path)
```

Comment:
Isn't `fm_path` undefined when `try_read_cache=False`?

Reply:
indeed, lifted it above the if block